src/coreclr/jit/instr.cpp

```diff
@@ -1096,19 +1209,69 @@ void CodeGen::inst_RV_TT_IV(instruction ins, emitAttr attr, regNumber reg1, GenT
     }
 }
 
+//------------------------------------------------------------------------
+// IsEmbeddedBroadcastEnabled: determine if embedded broadcast can be enabled
+//
+// Arguments:
+//    ins       -- The instruction being emitted
+//    op        -- The second operand of the instruction.
+//
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+bool CodeGenInterface::IsEmbeddedBroadcastEnabled(instruction ins, GenTree* op)
+{
+    if (!GetEmitter()->UseEvexEncoding())
+    {
+        return false;
+    }
+
+    bool IsEmbBroadcastCompatible = instIsEmbeddedBroadcastCompatible(ins);
+    if (!IsEmbBroadcastCompatible)
+    {
+        return false;
+    }
+
+    // Embedded broadcast can be applied when operands are in the following forms.
+    // 1. (contained)Broadcast -> ContainedNode
+    if (!op->isContained() || !op->OperIsHWIntrinsic())
+    {
+        return false;
+    }
+    bool IsEmbBroadcastEnabled = false;
+    switch (op->OperGet())
+    {
+        case GT_HWINTRINSIC:
+        {
+            if (op->isContained() && op->AsHWIntrinsic()->OperIsBroadcastScalar())
+            {
+                IsEmbBroadcastEnabled = true;
+            }
+            break;
+        }
+
+        default:
+            break;
+    }
+
+    // to enable embedded broadcast, we need 3 things,
+    // 1. embedded broadcast compatible intrinsics
+    // 2. proper forms on the intrinsic operands.
+    // 3. EVEX enabled.
+    return IsEmbBroadcastCompatible && IsEmbBroadcastEnabled;
```

Comment:
This can now be simplified down to:
```suggestion
    // To enable embedded broadcast, we need 3 things,
    // 1. EVEX enabled.
    // 2. Embedded broadcast compatible intrinsics
    // 3. A contained broadcast scalar node

    if (!GetEmitter()->UseEvexEncoding())
    {
        return false;
    }

    if (!instIsEmbeddedBroadcastCompatible(ins))
    {
        return false;
    }

    if (!op->isContained() || !op->OperIsHWIntrinsic())
    {
        return false;
    }
    
    return op->AsHWIntrinsic()->OperIsBroadcastScalar();
```